### PR TITLE
fix(InstallWizard): Fix exception when InstallWizard completed

### PR DESCRIPTION
**Pull request by BYK**
Since we don't have tests for the submit flow, we did not catch a regression from #13798 where the event handler was not bound to the class instance, causing an `this.setState` is not defined error when submitting the welcome form.

This patch fixes the issue. Will add tests to cover this soon after.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/app.jsx [hunk 1]
```diff
@@ -200,16 +200,13 @@ class App extends React.Component {
     e.stopPropagation();
   }
 
-  onConfigured() {
-    this.setState({needsUpgrade: false});
+  onConfigured = () => this.setState({needsUpgrade: false};
   }
```
Comment (dashed) [inline src/sentry/static/sentry/app/views/app.jsx]: this is a stray `}`

**Review by dashed — DISMISSED** [review]

File: src/sentry/static/sentry/app/views/app.jsx (hunk 1 above)
Comment (BYK) [reply src/sentry/static/sentry/app/views/app.jsx]: lol

Comment (BYK) [reply src/sentry/static/sentry/app/views/app.jsx]: that's why this was a draft :D

Comment (dashed) [reply src/sentry/static/sentry/app/views/app.jsx]: i should do that too. draft PRs 🤣

File: src/sentry/static/sentry/app/views/app.jsx [hunk 2]
```diff
@@ -200,16 +200,13 @@ class App extends React.Component {
     e.stopPropagation();
   }
 
-  onConfigured() {
-    this.setState({needsUpgrade: false});
-  }
+  onConfigured = () => this.setState({needsUpgrade: false});
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/app.jsx]: 👍

**Comment by BYK** [comment]
I have verified that this works locally. We'll need to add a dedicated acceptance test suite for the welcome page which I'll do later on.

**Review by lynnagara — APPROVED** [review]
Thanks for fixing. Would be great to have tests /snapshots later - it's easy to inadvertently break this page

**Pull request merged**